### PR TITLE
Formalize `blake2s` builtin + stdlib spec + RFC 7693 reference

### PR DESCRIPTION
**Pull request by Eduardogbg**
Lampe-side semantics for the Noir `blake2s` foreign builtin against
an RFC 7693 reference.

### What's here
- `Lampe/Crypto/Blake2s.lean` — concrete reference: IV constants,
  sigma permutations, `G` quarter-round, compression `F`, full
  variable-length hash with finalisation.
- `Lampe/Builtin/Crypto/Blake2s.lean` — `blake2s` total-pure builtin
  descriptor.
- `Lampe/Tests/Blake2s.lean` — KAT against the RFC 7693 reference
  test vectors.
- `stdlib/lampe/Stdlib/Hash/Mod.lean` — `blake2s_builtin_spec` giving
  the strong equality `r = Crypto.Blake2s.blake2sHash input`.

### About the spec shape
Noir's `pub fn blake2s<N>(input) {}` is `#[foreign(...)]` with an empty
body. The Lampe extractor drops empty-body foreign functions and
rewrites their call sites to direct `BuiltinCallRef`s, so every
extracted reference to `blake2s` becomes a `Builtin.blake2S` call.
One `_builtin_spec` is therefore the complete public API.

### CODE REVIEW

File: Lampe/Lampe/Crypto/Blake2s.lean [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+/-! ### Test vectors
+
+The first two vectors (`empty`, `abc`) are the canonical RFC 7693
+references; the latter two cover the block-boundary edge cases.
+Outputs are computed by Python's `hashlib.blake2s`, which wraps the
+BLAKE2 reference implementation (`libb2`). Regenerate with
+`scripts/blake2s_ref.py`. -/
+
+-- empty: RFC 7693 reference empty-input hash, input length = 0
+private def emptyIn : Array (BitVec 8) :=
```
Comment (kustosz) [inline Lampe/Lampe/Crypto/Blake2s.lean]: These are pointlessly duplicated, please remove here or in tests.

**Review by kustosz — APPROVED** [review]
